frontend/sozisel-app/src/components/PresenterSession/EventsTimline/EventsTimeline.tsx

```diff
@@ -0,0 +1,220 @@
+    if (!endSessionError) {
+      history.push(`result/summary`);
+    }
+  };
+
+  useEffect(() => {
+    const lastEventIdx = launchedEvents.length - 1;
+    if (lastEventIdx >= 0) {
+      const lastEvent = launchedEvents[launchedEvents.length - 1];
+      const lastEventStartTime = moment.tz(
+        lastEvent.insertedAt,
+        "Europe/Warsaw"
+      );
+      const currentTime = Math.floor(
+        moment
+          .duration(moment.tz("Europe/Warsaw").diff(lastEventStartTime))
+          .asSeconds()
+      );
+      if (currentTime < events[lastEventIdx].eventData.durationTimeSec) {
+        setActiveEventIdx(lastEventIdx);
+        setActiveEventId(lastEvent.id);
+        setActiveEventCurrentSec(
+          events[lastEventIdx].eventData.durationTimeSec - currentTime
+        );
+      } else {
+        setActiveEventIdx(lastEventIdx + 1);
+      }
+    }
+  }, [activeEventIdx, events, launchedEvents]);
+
+  const [launchEventMutation, { error: launchEventError }] =
+    useLaunchEventMutation();
+
+  const onNextEvent = async () => {
+    const {
+      id,
+      eventData: { targetPercentageOfParticipants },
+    } = events[activeEventIdx];
+    await launchEventMutation({
+      variables: {
+        broadcast: targetPercentageOfParticipants === 100,
+        eventId: id,
+        sessionId,
+        targetParticipants: getRandomParticipants(
+          participants,
+          targetPercentageOfParticipants
+        ),
+      },
+    });
+    if (!launchEventError && activeEventIdx < events.length) {
+      setActiveEventCurrentSec(
+        events[activeEventIdx].eventData.durationTimeSec
+      );
+      setActiveEventId(events[activeEventIdx].id);
+    }
+  };
+
+  return (
+    <Paper elevation={2} className="EventsTimeline">
+      <div className="header">
+        <CalendarTodayIcon className="icon" />
+        <Typography>
+          {t("components.PresenterSession.EventsTimeline.header")}
+        </Typography>
+      </div>
+      <Stepper activeStep={activeEventIdx} alternativeLabel className="stepper">
+        {events.map(({ name, id, startMinute }, idx) => (
+          <Step key={id}>
+            <StepLabel className="label">
+              {name + " - "}
+              <b>
+                {t("components.PresenterSession.EventsTimeline.startMinute", {
+                  value: startMinute,
+                })}
+              </b>
+              {idx === activeEventIdx && !activeEventId && (
+                <IconButton
+                  color="primary"
+                  className="startEventButton"
+                  onClick={onNextEvent}
+                >
+                  <PlayCircleFilledIcon />
+                </IconButton>
+              )}
+            </StepLabel>
+          </Step>
+        ))}
+      </Stepper>
+      {activeEventIdx < events.length && (
+        <div className="eventDetails">
+          {!activeEventId &&
+            events[activeEventIdx].eventData.__typename === "Quiz" && (
```

Comment:
I guess each new event will have to have that typename checking, extract it to another component that on its own will render appropriate components. This file is already to large.